httpbin/helpers.py

```diff
@@ -412,3 +421,152 @@ def get_request_range(request_headers, upper_bound):
+    path = path.rstrip("/")
+
+    # Method used from this SO response: http://stackoverflow.com/a/600612
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+def _generate_filename(ctype=None):
+    """Returns a unique file name with an extension corresponding to given ctype
+    If ctype is None, return a file name with no extension"""
+    name = str(uuid.uuid4())
+
+    if ctype is None:
+        return name
+    elif ctype == "plain/text":
+        return name + ".txt"
+    elif ctype == "application/json":
+        return name + ".json"
+    else:
+        return name
+
+
+def create_file(path, content, filename=None, ctype=None):
+    """Creates a file at 'path' with 'content'"""
+    # return empty string if the file already exists
+    filename = '' if filename is None else filename
+    if os.path.isfile(os.path.join(path,filename)):
+        return ''
+
+    _mkdir_p(path)
+
+    if not filename:
+        filename = _generate_filename(ctype)
+    path = os.path.join(path, filename)
+
+    fo = open(path, 'w+')
+    fo.write(content)
+    fo.close()
+
+    return path
+
+def get_all_files_in_dir(dir_path, created_since, limit):
+    """Returns at max 'limit' number of files from the given directory
+    If created_since is passed, it will only return the files that were created
+    in last created_since seconds"""
+
+    return_files = []
+    now = dt.datetime.now()
+
+    for root, dir, files in os.walk(dir_path):
+        for fname in files:
+            full_path = os.path.join(root,fname)
+            if os.path.isfile(full_path):
+                stat = os.stat(full_path)
+                ctime = dt.datetime.fromtimestamp(stat.st_ctime)
+
+                if created_since is not None:
+                    if ctime >= now - int(created_since):
+                        return_files.append(full_path)
+                else:
+                    return_files.append(full_path)
+
+                # only return up to limit number of files
+                if len(return_files) >= limit:
+                    return return_files
+
+    return return_files
+
+
+## MIME
+def encode_multipart(fields, files, boundary=None):
+    """
+    Taken from: http://code.activestate.com/recipes/578668-encode-multipart-form-data-for-uploading-files-via/
+
+    Encode dict of form fields and content of files as multipart/form-data.
+    Return tuple of (body_string, headers_dict). Each value in files is a full path
+    to the file.  If not specified, tries to guess mime type or uses
+    'application/octet-stream')
+    """
+    _BOUNDARY_CHARS = string.digits + string.ascii_letters
+
+    if boundary is None:
+        boundary = ''.join(random.choice(_BOUNDARY_CHARS) for i in range(30))
+
+    def escape_quote(s):
+        return s.replace('"', '\\"')
+
+    lines=[]
+
+    for name, value in fields.items():
```

Comment:
Remove `fields`.  We don't need that for its current use case - send multiple files as multipart/form-data